store/boltdb/boltdb.go

```diff
@@ -50,39 +70,79 @@ func New(endpoints []string, options *store.Config) (store.Store, error) {
 	}
 
 	dir, _ := filepath.Split(endpoints[0])
-	if err := os.MkdirAll(dir, 0750); err != nil {
+	if err = os.MkdirAll(dir, 0750); err != nil {
 		return nil, err
 	}
 
-	var boltOptions *bolt.Options
-	if options != nil {
+	if !options.TransientConnection {
```

Comment:
ofcourse if we decide to flip the meaning of this variable, all the current usage must flip.